process/smartContract/process.go

```diff
@@ -2795,6 +2800,47 @@ func (sc *scProcessor) processSimpleSCR(
 	return nil
 }
 
+// CheckBuiltinFunctionIsExecutable validates the builtin function arguments and tx fields without executing it
+func (sc *scProcessor) CheckBuiltinFunctionIsExecutable(expectedBuiltinFunction string, tx data.TransactionHandler) error {
+	if check.IfNil(tx) {
+		return process.ErrNilTransaction
+	}
+
+	functionName, arguments, err := sc.argsParser.ParseCallData(string(tx.GetData()))
+	if err != nil {
+		return err
+	}
+
+	if expectedBuiltinFunction != functionName {
+		return process.ErrBuiltinFunctionMismatch
+	}
+
+	gasProvided, err := sc.prepareGasProvided(tx)
+	if err != nil {
+		return err
+	}
+
+	builtinFunc, err := sc.builtInFunctions.Get(functionName)
+	if err != nil {
+		return err
+	}
+
+	// cast the function to the expected checker type
+	executableChecker, ok := builtinFunc.(ExecutableChecker)
```

Comment:
in the scProcessor constructor you could set up a container with all the builtInFunctions which are executable. and here you do not need to make the cast.
Or add a public function in builtInFunctionsContainer to get the executableChecker from there.

Reply:
added a TODO for a refactor.
I don't see a clean way to do it right now and avoid the typecast.

Reply:
created a map for the executable checkers